src/picongpu/include/particles/InitFunctors.hpp

```diff
@@ -91,16 +90,16 @@ struct CreateDensity
     /* add interface for compile time interface validation*/
     typedef startPosition::IFunctor<UserPositionFunctor> PositionFunctor;
 
-    template<typename T_StorageTuple>
-    HINLINE void operator()(
-                            T_StorageTuple& tuple,
-                            const uint32_t currentStep
-                            )
+    HINLINE void operator()( const uint32_t currentStep )
     {
-        auto speciesPtr = tuple[SpeciesName()];
+        DataConnector &dc = Environment<>::get().DataConnector();
+        auto speciesPtr = dc.get< SpeciesType >( SpeciesType::FrameType::getName(), true );
```

Comment:
species name is already avail: please use `dc.get< SpeciesType >( SpeciesName(), true );`